src/ring/logger/messages.clj

```diff
@@ -84,7 +90,10 @@
    title time-str status-str]
   (let [log-message (str title
                          request-method " "
-                         uri  (if query-string (str "?" query-string))
+                         (if (and query-string
+                                  (:log-query-string? options))
+                           (str  uri "?" query-string)
+                           uri)
```

Comment:
same thing about extracting `uri` and turn `if` into `when`

Reply:
Done